### DISCUSSION TITLE
Could not find a strategy to sign you up with. Did you fill out the form correctly? in Angular SPA

### DISCUSSION THREAD

**Discussion by aaditya-madala** [ory/kratos#2022] [Q&A] [answered]
I am trying to use Ory Kratos with an Angular SPA. I am following the guide for client side apps here: https://www.ory.sh/kratos/docs/self-service#browser-flows-for-client-side-apps-single-page-apps-reactjs-angularjs-nextjs-. 

- A copy of some of the elements of the app are here: https://github.com/aaditya-madala/kratos-poc-question. 
- For the identity schema we are using the quickstart version: https://github.com/aaditya-madala/kratos-poc-question/blob/main/kratos-config/config/kratos/email-password/identity.schema.json
- For kratos.yml I am using the following where the app url is 127.0.0.1:4200 (local ng serve'd angular app): https://github.com/aaditya-madala/kratos-poc-question/blob/main/kratos-config/config/kratos/email-password/kratos.yml
- Using the quickstart and quickstart-postgres .yml files I was able to run `docker-compose -f quickstart.yml -f quickstart-postgres.yml up --build --force-recreate` and successfully run kratos locally.
- I am able to hit the http://127.0.0.1:4433/self-service/registration/browser kratos endpoint to get a flow id and flow object with the nodes information.
- After which I get a valid response:

![image](https://user-images.githubusercontent.com/14017280/144644283-44f1ef08-9dc4-48d0-ba2f-ea48bd2f2ef1.png)

![image](https://user-images.githubusercontent.com/14017280/144644365-cebb280d-39eb-455d-8ae3-603d5a6aeb99.png)

- Sending a response to the registration flow endpoint with the form data:

![image](https://user-images.githubusercontent.com/14017280/144647044-05988d45-5401-4df1-a1ff-782438a7fff9.png)

![image](https://user-images.githubusercontent.com/14017280/144647260-a92d63f9-7cb4-45bb-b44e-6871644ebc03.png)

I get the above error when I submit the registration form details. Do you know what I am missing in the registration flow for SPA?

 

>

**Comment by aeneasr** [comment, marked as answer]
Make sure you send the data in the correct format. So if it is JSON encoded send it with `Content-Type: application/json` and the form encoding otherwise!

**Comment by chihabaigot101** [comment]
![image](https://user-images.githubusercontent.com/121164719/212940221-9f1faf11-ab72-4e55-a44f-a6da8e5f0a16.png)
i traced the request, the data is being sent (debugged using formdata encoding) tried every encoding type but to no aveil

**Reply by MJEND7** [reply]
I am facing a similar issue also
